### PR TITLE
add transformers example

### PR DESCRIPTION
**Pull request by aminsaied**
Adding a few transformers examples for finetuning glue task.

Couple issues:
- [ ] readme does not update because I have not named the files job.py. You will see from the directory structure why.
- [ ] The autogenerated README - if it did pick up these files - would incorrectly label the compute target, as one of them uses multiple computes, both CPU and GPU. Its a bit of a corner case, but maybe worth calling out

### CODE REVIEW

**Review by lostmygithubaccount — CHANGES_REQUESTED** [review]
- [x] add `job` to workflow names
- [x] update contributing naming conventions accordingly
- [x] edit `readme.py` to pickup any `*job*.py` when globbing workflows

**Comment by lostmygithubaccount** [comment]
sry for hijacking your branch, wanted to get this in tonight

**Comment by lostmygithubaccount** [comment]
ah i think the examples are missing run.wait_for_completion()

**Comment by lostmygithubaccount** [comment]
missing something: https://ml.azure.com/experiments/id/0a183c72-5650-44d6-9ee4-3f7dced031be/runs/transformers-glue-finetuning-sku-comparison_1605159810_d2515963?wsid=/subscriptions/6560575d-fa06-4e7d-95fb-f962e74efd7a/resourcegroups/azureml-examples/workspaces/default&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#outputsAndLogs

**Comment by lostmygithubaccount** [comment]
strange failure: https://ml.azure.com/experiments/id/58de69a0-ccfa-4f8d-a1eb-69764a3a21d3/runs/transformers-glue-finetuning_1605160966_a062e17e?wsid=/subscriptions/6560575d-fa06-4e7d-95fb-f962e74efd7a/resourcegroups/azureml-examples/workspaces/default&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#outputsAndLogs

File: workflows/train/transformers/glue/1-aml-finetune-job.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    print(f"Create Azure ML Environment from {pip_requirements_path}")
+
+    if use_gpu:
+
+        env_name = "transformers-gpu"
+        env = Environment.from_pip_requirements(
+            name=env_name, file_path=pip_requirements_path,
+        )
+        env.docker.base_image = (
+            "mcr.microsoft.com/azureml/intelmpi2018.3-cuda10.0-cudnn7-ubuntu16.04"
```
Comment (lostmygithubaccount) [inline workflows/train/transformers/glue/1-aml-finetune-job.py:70]: any downside to using this base image for CPU and GPU for simplicity?

File: workflows/train/transformers/glue/2-aml-comparison-of-sku-job.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    print(f"Create Azure ML Environment from {pip_requirements_path}")
+
+    if use_gpu:
+
+        env_name = "transformers-gpu"
+        env = Environment.from_pip_requirements(
+            name=env_name, file_path=pip_requirements_path,
+        )
+        env.docker.base_image = (
+            "mcr.microsoft.com/azureml/intelmpi2018.3-cuda10.0-cudnn7-ubuntu16.04"
```
Comment (lostmygithubaccount) [inline workflows/train/transformers/glue/2-aml-comparison-of-sku-job.py:59]: same here

File: workflows/train/transformers/glue/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,7 @@
+git+git://github.com/huggingface/transformers.git
```
Comment (lostmygithubaccount) [inline workflows/train/transformers/glue/requirements.txt:1]: why install from github? seems unstable

File: workflows/train/transformers/glue/src/finetune_glue.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        callbacks=[AzureMLCallback()],
+        train_dataset=encoded_dataset_train,
+        eval_dataset=encoded_dataset_eval,
+        tokenizer=tokenizer,
+        compute_metrics=compute_metrics,
+    )
+
+    print("Training...")
+
+    run = Run.get_context()  # get handle on Azure ML run
```
Comment (lostmygithubaccount) [inline workflows/train/transformers/glue/src/finetune_glue.py:82]: should use mlflow for logging

File: workflows/train/transformers/glue/requirements.txt (hunk 3 above)
Comment (aminsaied) [reply workflows/train/transformers/glue/requirements.txt:1]: latest examples from transformers are only available directly from source.

File: workflows/train/transformers/glue/1-aml-finetune-job.py (hunk 1 above)
Comment (aminsaied) [reply workflows/train/transformers/glue/1-aml-finetune-job.py:70]: To be honest, you wouldn't want to run these examples on anything other than a gpu. What do you think to only providing the GPU-enabled environment?

File: workflows/train/transformers/glue/2-aml-comparison-of-sku-job.py (hunk 2 above)
Comment (aminsaied) [reply workflows/train/transformers/glue/2-aml-comparison-of-sku-job.py:59]: As above...

File: workflows/train/transformers/glue/src/finetune_glue.py (hunk 4 above)
Comment (aminsaied) [reply workflows/train/transformers/glue/src/finetune_glue.py:82]: huggingface has a specific callback for azureml which I wanted to demo...
